thetis/configuration.py

```diff
@@ -166,6 +167,23 @@ def default_value_repr(self):
         return 'Function'
 
 
+class FiredrakeExpression(TraitType):
+    default_value = None
+    info_text = 'a UFL expression, Firedrake Constant or Function'
+
+    def validate(self, obj, value):
+        if isinstance(value, (Constant, Function, ufl.core.operator.Operator)):
```

Comment:
I'm not sure if `ufl.core.operator.Operator` is the best way of checking whether `value` is a UFL expression.

Reply:
I think the most generic would be:

    if isinstance(value, ufl.core.expr.Expr):

this would also cover things like `uv[0]` or `0*u` (where u is a Function). Only small downside I can see is that this allow TestFunction and TrialFunction as well. We could also consider to split this into a `FiredrakeScalarExpression` and `FiredrakeVectorExpression` and use `ufl.checks.is_ufl_scalar()`.

Reply:
Good idea, implemented.